src/chains/algorand_1/algoChainState.ts

```diff
@@ -65,8 +67,11 @@ export class AlgorandChainState {
   public async connect(): Promise<void> {
     try {
       if (!this._algoClient) {
-        const { token, server, port } = this.getAlgorandConnectionSettings()
-        this._algoClient = new algosdk.Algod(token, server, port)
+        const { url, endpoint } = this.selectEndpoint()
```

Comment:
I think this would be a lot cleaner if you did
```        
        const { endpoint } = this.selectEndpoint()
        this._activeEndpoint = endpoint
        const { token, url, port } = this.getAlgorandConnectionSettingsForEndpoint(endpoint)
        this._algoClient = new algosdk.Algod(token, url, port)
```
Then have getAlgorandConnectionSettingsForEndpoint pull things apart

Reply:
done